### ISSUE TITLE
Add MacOS headers to get bdwgc to compile

### ISSUE THREAD

**Issue by plajjan** [ziglang/zig#18257]
### Zig Version

0.12.0-dev.1536+6b9f7e26c

### Steps to Reproduce and Observed Behavior

The following header files are missing on Macos, which I need to compile bdwgc (https://github.com/actonlang/bdwgc/actions/runs/7168073811/job/19515450726):
- mach-o/getsect.h
- mach/exception.h
- CoreFoundation/* (CoreFoundation/CoreFoundation includes all the other header files, so they come as one group)

### Expected Behavior

I want it to compile please! :)

**Comment by nektro** [comment]
cc @kubkon

**Comment by kubkon** [comment]
Mach headers - sure thing, we should upload them, framework headers are out of scope however. For frameworks, headers and dylib stubs, you will need to obtain on your own. An example of how this can be done in accordance with Apple eula is in https://github.com/hexops/xcode-frameworks

**Comment by plajjan** [comment]
@kubkon thx! Are framework headers out of scope due to licensing or just a principal decision not to include those kinds of headers in zig?

**Comment by kubkon** [comment]
> @kubkon thx! Are framework headers out of scope due to licensing or just a principal decision not to include those kinds of headers in zig?

Licensing and the sheer size of the framework headers and stubs. Current plan is for a 3rd party to offer it as a Zig package.

**Comment by plajjan** [comment]
@kubkon okok, makes sense. We managed to reduce the need for that CoreFoundation header (now only needed for IOS builds and I'm hopeful to remove that too), so the remaining mach / mach-o headers are the only remaining things missing! :)

**Comment by plajjan** [comment]
FWIW, the Boehm-Demers-Weiser Garbage Collector, perhaps the most famous GC around, now has experimental support for using the zig build system, see https://github.com/ivmai/bdwgc/commit/2666e3e54369309f04073063521a92e4bd77333f (ping @andrewrk, maybe you're interested :)). Thanks a lot to @ivmai (the BDWGC maintainer) who helped make this happen, awesome work!

As we all know, zig is awesome at cross-compile and so it also unlocked cross-compilation testing (although not run-time testing) of the BDWGC for CI testing, https://github.com/ivmai/bdwgc/actions/runs/7377734816. Very cool! Fixing this PR would help enabling cross-compilation builds targeting macos there as well! :)

**Comment by alexrp** [comment]
Seems like there's nothing left to be done here, considering bdwgc has had Zig build support for a while?

Let me know if I'm mistaken, though, and I can reopen.

**Comment by linusg** [comment]
I don't think this is fixed, they just work around it with build flags: https://github.com/bdwgc/bdwgc/blob/44fbb3b9e4c55d9fa2e134948ff542fe3a774e80/build.zig#L468-L475
